plugins/woocommerce-blocks/assets/js/blocks/product-collection/constants.ts

```diff
@@ -71,6 +71,7 @@ export const DEFAULT_ATTRIBUTES: Partial< ProductCollectionAttributes > = {
 		columns: 3,
 		shrinkColumns: true,
 	},
+	includeInQueryContext: [ 'collection', 'id' ],
```

Comment:
As we discussed on a call, let's pass `collection` and `id` directly to the `getEntityRecords`. But of course it's good to keep them here for demo purposes and as an example! 🚀